src/resolve-image.js

```diff
@@ -0,0 +1,121 @@
+  const registry = parsedImage.registry
+  let registryUrl
+  if (registry) {
+    registryUrl = `https://${registry}/v2/`
+  } else {
+    registryUrl = DEFAULT_REGISTRY
+  }
+  return registryUrl
+}
+
+const getAuthInfo = async function (image) {
+  try {
+    const registryUrl = getRegistryUrl(image)
+    // Returns 401 response with auth requirements for registry
+    debug(`fetching auth requirements from ${registryUrl}`)
+    const realmResp = await axios.get(registryUrl, {
+      validateStatus: (status) => { return status === 401 }
+    })
+    const authInfo = {}
+    const realmInfo = realmResp.headers['www-authenticate'] // Realm and service params
+    if (realmInfo) {
+      const re = /Bearer realm="(.*)",service="(.*)"/i
+      const found = realmInfo.match(re)
+      if (found) {
+        [, authInfo.realm, authInfo.service] = found
+      }
+    } else {
+      throw new Error('Failed to parse realm and service info from registry.')
+    }
+    return authInfo
+  } catch (err) {
+    debug(err)
+    throw new Error('Failed to get authentication info from registry')
+  }
+}
+
+const getAuthToken = async function (image, authInfo) {
+  try {
+    const { namespace, repository } = parseImage(image)
+    const tokenOpts = {
+      params: {
+        service: authInfo.service,
+        scope: `repository:${namespace || 'library'}/${repository}:pull`
+      }
+    }
+
+    debug(`fetching auth token from ${authInfo.realm}...`)
+    const tokenResp = await axios.get(authInfo.realm, tokenOpts)
+    const token = tokenResp.data.token
+    if (!token) {
+      throw new Error("Can't get authentication token from registry.")
+    }
+    const tokenInfo = {
+      ...authInfo,
+      token: token
+    }
+    return tokenInfo
+  } catch (err) {
+    debug(err)
+    throw new Error('Failed to get authentication token from registry.')
+  }
+}
+
+const getManifestResp = async function (image, token) {
+  try {
+    const manifestOptions = {
+      headers: {
+        Authorization: `Bearer ${token.token}`,
+        // Required header for manifest endpoint - see https://github.com/docker/distribution/issues/1565
+        Accept: 'application/vnd.docker.distribution.manifest.v2+json'
+      }
+    }
+
+    const { namespace, repository, tag } = parseImage(image)
+    const registryUrl = getRegistryUrl(image)
+
+    // Build url for manifest request - see https://docs.docker.com/registry/spec/api/
+    let manifestUrl = `${registryUrl}${namespace || 'library'}/${repository}/manifests/${tag}`
+    debug(`fetching manifest info from ${manifestUrl} with options ${JSON.stringify(manifestOptions, null, 2)}`)
+    return await axios.get(manifestUrl, manifestOptions)
+  } catch (err) {
+    debug(err)
+    throw new Error('Unable to get manifest info from registry.')
+  }
+}
+
+const resolveImage = async function (image) {
+  if (image.includes('@sha256:')) {
+    return image
+  }
+
+  debug(`fetching auth info for image ${image}`)
```

Comment:
This is a small detail, but our log messages typically use the form of `message. name=value otherName=otherValue`. So this might be `fetching auth info for image. image=${image}`